woocommerce/woocommerce-gutenberg-products-block
# Document and update naming of Block Asset Handles

This PR establishes the pattern of naming block handles as `wc-{blockname}-block`, and other scripts as `wc-blocks-{script}` to avoid conflicts.

To help map handles -> built scripts I've also made script filenames match. I did not do this for the blocks themselves because it caused some breakage in Atomic Blocks and complicates how blocks are setup in webpack.

Closes #2660

### How to test the changes in this Pull Request:

- Tests should pass
- Check existing block types still show up in the editor and can be inserted without regression.
- Console should be error-free
- Specifically test single-product block works to verify Atomic Blocks are not broken

### Dev Note

Some script handles have been changed in this version for consistency. It is unlikely that these handles are being used directly by 3rd parties because they are only used within the blocks, however for reference:

- `wc-block-vendors-style` -> `wc-blocks-vendors-style`
- `wc-block-editor` -> `wc-blocks-editor-style`
- `wc-block-style` -> `wc-blocks-style`
- `wc-vendors` -> `wc-blocks-vendors`
- `wc-shared-context` -> `wc-blocks-shared-context`
- `wc-shared-hocs` -> `wc-blocks-shared-hocs`

In addition, all block script handles now have the `-block` suffix.

### Changelog

> Made script and style handles consistent.
